src/vs/workbench/contrib/notebook/browser/viewParts/notebookEditorStickyScroll.ts

```diff
@@ -54,26 +59,56 @@ export class ToggleNotebookStickyScroll extends Action2 {
 export class NotebookStickyLine extends Disposable {
 	constructor(
 		public readonly element: HTMLElement,
+		public readonly foldingIcon: StickyFoldingIcon,
+		public readonly header: HTMLElement,
 		public readonly entry: OutlineEntry,
 		public readonly notebookEditor: INotebookEditor,
 	) {
 		super();
-		this._register(DOM.addDisposableListener(this.element, DOM.EventType.CLICK, () => {
+		// click the header to focus the cell
+		this._register(DOM.addDisposableListener(this.header, DOM.EventType.CLICK || TouchEventType.Tap, () => {
 			this.focusCell();
 		}));
+
+		// click the folding icon to fold the range covered by the header
+		this._register(DOM.addDisposableListener(this.foldingIcon.domNode, DOM.EventType.CLICK || TouchEventType.Tap, () => {
+			if (this.entry.cell.cellKind === CellKind.Markup) {
+				const currentFoldingState = (this.entry.cell as MarkupCellViewModel).foldingState;
+				this.toggleFoldRange(currentFoldingState);
+			}
+		}));
+
+		// folding icon hovers
+		// this._register(DOM.addDisposableListener(this.element, DOM.EventType.MOUSE_OVER, () => {
+		// 	this.foldingIcon.setVisible(true);
+		// }));
+		// this._register(DOM.addDisposableListener(this.element, DOM.EventType.MOUSE_OUT, () => {
+		// 	this.foldingIcon.setVisible(false);
+		// }));
+
+	}
+
+	private toggleFoldRange(currentState: CellFoldingState) {
+		const foldingController = this.notebookEditor.getContribution('workbench.notebook.foldingController') as FoldingController;
```

Comment:
We can do `this.notebookEditor.getContribution<FoldingController>(FoldingController.id)